From 91f9b02553371a085fed75b86f8a9bcac6cfe36a Mon Sep 17 00:00:00 2001
From: Derek Lesho <dlesho@codeweavers.com>
Date: Mon, 30 Nov 2020 11:56:48 -0500
Subject: [PATCH] mf: Add invalid connect method test.

Nikolay stripped out this test in his updated version of the patchset, which is fine.  But I think it's useful to have it somewhere.

Signed-off-by: Derek Lesho <dlesho@codeweavers.com>
---
 dlls/mf/tests/mf.c | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/dlls/mf/tests/mf.c b/dlls/mf/tests/mf.c
index 8272064466f..32ae84b837b 100644
--- a/dlls/mf/tests/mf.c
+++ b/dlls/mf/tests/mf.c
@@ -1824,6 +1824,34 @@ static void test_topology_loader(void)
             LOADER_TODO,
         },
 
+        {
+            /* MP3 -> PCM */
+            &MFMediaType_Audio,
+            {
+                {
+                  { &MF_MT_SUBTYPE, WAVE_FORMAT_MPEGLAYER3 },
+                  { &MF_MT_AUDIO_NUM_CHANNELS, 2 },
+                  { &MF_MT_AUDIO_SAMPLES_PER_SECOND, 44100 },
+                  { &MF_MT_AUDIO_AVG_BYTES_PER_SECOND, 16000 },
+                  { &MF_MT_AUDIO_BLOCK_ALIGNMENT, 1 },
+                }
+            },
+            {
+                {
+                  { &MF_MT_SUBTYPE, WAVE_FORMAT_PCM },
+                  { &MF_MT_AUDIO_NUM_CHANNELS, 1 },
+                  { &MF_MT_AUDIO_SAMPLES_PER_SECOND, 44100 },
+                  { &MF_MT_AUDIO_AVG_BYTES_PER_SECOND, 44100 },
+                  { &MF_MT_AUDIO_BLOCK_ALIGNMENT, 1 },
+                  { &MF_MT_AUDIO_BITS_PER_SAMPLE, 8 },
+                }
+            },
+
+            MF_CONNECT_ALLOW_DECODER &~ MF_CONNECT_ALLOW_CONVERTER,
+            MF_E_INVALIDMEDIATYPE,
+            LOADER_TODO,
+        },
+
         {
             /* MP3 -> PCM */
             &MFMediaType_Audio,
-- 
2.29.2

